### PR TITLE
Add git revision id

### PR DESCRIPTION
**Pull request by Deren-Liao**
to log entries if the source context file exists and contains git repo
information.

### CODE REVIEW

**Comment by Deren-Liao** [comment]
Hi there,

As I clicked the sign url, it told me I am a *Googlers do not need to sign
a CLA.*

**Comment by jskeet** [comment]
I suspect the reason is that the commit has Charlie-ET as the author, instead of the Deren-Liao user. I suspect you just need to amend the email address used for that commit - and probably change your defaults for your local clone too.

**Comment by Deren-Liao** [comment]
Yes, that's the reason. I did some test with another account.

**Comment by Deren-Liao** [comment]
I fixed the CLA problem.  Some tests failed.  I don't think it's related to my change.  How may I let the test try again?

**Comment by jskeet** [comment]
Hmm... the BigQuery tests appear to have hung. Have restarted it.

File: apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs [hunk 1]
```diff
@@ -289,6 +292,23 @@ private LogEntry BuildLogEntry(LoggingEvent loggingEvent)
...
+        private void TryAddGitRevisionId(Dictionary<string, string> labels)
+        {
+            string gitId = null;
+            try
+            {
+                gitId = SourceContext.AppSourceContext?.Git?.RevisionId;
+            }
+            catch (Exception ex) when (ex is System.Security.SecurityException || ex is InvalidProtocolBufferException
+                || ex is InvalidJsonException || ex is UnauthorizedAccessException)
+            { }
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs]: I'd reformat this as one type per line for readability. And could we at least have a comment in the catch body explaining why we're not doing anything?

File: apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs [hunk 2]
```diff
@@ -289,6 +292,23 @@ private LogEntry BuildLogEntry(LoggingEvent loggingEvent)
...
+
+            if (!String.IsNullOrWhiteSpace(gitId))
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs]: I'd move this inside the catch block - then you don't need the variable to be declared first. The Add call won't throw any of the caught exceptions, so it won't change behaviour.

File: apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs [hunk 3]
```diff
@@ -289,6 +292,23 @@ private LogEntry BuildLogEntry(LoggingEvent loggingEvent)
...
+            {
+                labels.Add(SourceContext.GitRevisionIdLogLabel, gitId);
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs]: I think I missed where `GitRevisionIdLogLabel` was added to `SourceContext`. Is that definitely the right place for it? It feels like a logging concern more than one of `SourceContext` itself...

File: apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs (hunk 2 above)
Comment (Deren-Liao) [inline apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs]: Done.

File: apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs (hunk 1 above)
Comment (Deren-Liao) [inline apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs]: Done.

File: apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs (hunk 3 above)
Comment (Deren-Liao) [inline apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs]: My perception is that It's a log label for source context. It's related to both.  I feel it's acceptable to put it in either here or in the Google.Cloud.Logging.V2.dll.

**Comment by Deren-Liao** [comment]
Thank you Jon for all the help and input.

**Pull request merged**